webui/react/src/pages/F_ExpList/F_ExperimentList.tsx

```diff
@@ -81,6 +98,28 @@ const F_ExperimentList: React.FC<Props> = ({ project }) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [page, sortString]);
 
+  useEffect(() => {
+    // useSettings load the default value first, and then load the data from DB
+    // use this useEffect to re-init the correct useSettings value when settings.filterset is changed
+    let ignore = false;
+
+    if (!ignore) {
+      const formSetValidation = IOFilterFormSet.decode(JSON.parse(settings.filterset));
+      if (isLeft(formSetValidation)) {
+        handleError(formSetValidation.left, {
+          publicSubject: 'Unable to initialize filterset from settings',
+        });
+      } else {
+        const formset = formSetValidation.right;
+        formStore.init(formset);
+      }
+    }
+
+    return () => {
+      ignore = true;
+    };
+  }, [settings.filterset]);
```

Comment:
`ignore` isn't necessary here. That pattern is used when we're making an asynchronous call and it's possible that by the time the asynchronous call is complete, the closure in the effect is state. Here, we immediately check `ignore` after setting it -- it will always be false.
```suggestion
  useEffect(() => {
    // useSettings load the default value first, and then load the data from DB
    // use this useEffect to re-init the correct useSettings value when settings.filterset is changed
    const formSetValidation = IOFilterFormSet.decode(JSON.parse(settings.filterset));
    if (isLeft(formSetValidation)) {
      handleError(formSetValidation.left, {
        publicSubject: 'Unable to initialize filterset from settings',
      });
    } else {
      const formset = formSetValidation.right;
      formStore.init(formset);
    }
  }, [settings.filterset]);
```